File: aurora-scala/src/main/scala/net/gree/aurora/scala/application/AuroraShardingConfigLoadService.scala
package net.gree.aurora.scala.application

import java.io.File
import net.gree.aurora.application.{AuroraShardingConfigLoadService => JAuroraShardingConfigLoadService, AuroraShardingConfigLoadServiceFactory}
import org.sisioh.config.Configuration
import scala.util.Try

trait AuroraShardingConfigLoadService {

  def loadFromConfigFile(configFile: File): Try[Repositories]

  def loadFromConfig(config: Configuration): Try[Repositories]

}

object AuroraShardingConfigLoadService {

  private[scala] def apply(underlying: JAuroraShardingConfigLoadService): AuroraShardingConfigLoadService =
    new AuroraShardingConfigLoadServiceImpl(underlying)

  def apply(): AuroraShardingConfigLoadService =
    apply(AuroraShardingConfigLoadServiceFactory.create())

}

File: aurora-core/src/main/java/net/gree/aurora/application/AuroraShardingConfigLoadService.java
package net.gree.aurora.application;

import com.typesafe.config.Config;
import org.sisioh.dddbase.utils.Try;

import java.io.File;

/**
 * 設定ファイルを読み込み、{@link Repositories} を生成するサービス。
 */
public interface AuroraShardingConfigLoadService {

    /**
     * 設定ファイルから読み込み、{@link Repositories} を生成する。
     *
     * @param configFile 設定ファイル
     * @return Tryでラップされた{@link Repositories}
     */
    Try<Repositories> loadFromConfigFile(File configFile);

    /**
     * 設定ファイルから読み込み、{@link Repositories} を生成する。
     *
     * @param config 設定ファイル
     * @return Tryでラップされた{@link Repositories}
     */
    Try<Repositories> loadFromConfig(Config config);

}
